### PR TITLE
Implement domain security email page with stubs for EPP

### PR DESCRIPTION
**Pull request by rachidatecs**
## 🗣 Description ##

- Stubs for EPP GET and POST
- Add security email form
- Add security email view
- Add security email template
- Routing
- Testing
- Remove security email from DomainInformation model and run migrations*

## 💭 Motivation and context ##

Completes[ 516](https://github.com/cisagov/getgov/issues/516)

 *NOTE: Security email will be done via EPP instead of DomainInformation.

### CODE REVIEW

File: src/registrar/views/domain.py [hunk 1]
```diff
@@ -96,6 +96,55 @@ def form_valid(self, formset):
...
+        """The form is valid, call setter in model."""
+
+        # Set the security email from the from
+        try:
+            new_email = form.cleaned_data["security_email"]
+        except KeyError:
+            # no server information in this field, skip it
+            pass
+        domain = self.get_object()
+        domain.set_security_email(new_email)
```
Comment (abroddrick) [inline src/registrar/views/domain.py:139]: You used new_email here but if it fails on the try new_email will have never been set.   In the except you can set new_email to none if that is the intention and requirement if a security email doesn't exist. Or just use a else after the except and only set the domain email if there is in fact an email available to use.

File: src/registrar/views/domain.py [hunk 2]
```diff
@@ -96,6 +96,55 @@ def form_valid(self, formset):
...
+        if form.is_valid():
+            # there is a valid email address in the form
+            return self.form_valid(form)
+        else:
+            return self.form_invalid(form)
+
+    def form_valid(self, form):
+        """The form is valid, call setter in model."""
+
+        # Set the security email from the from
```
Comment (abroddrick) [inline src/registrar/views/domain.py:132]:
```suggestion
        # Set the security email from the form
```
typo

**Review by abroddrick — COMMENTED** [review]
Didn't do a full review, just adding a couple comments.

File: src/registrar/views/domain.py (hunk 1 above)
Comment (rachidatecs) [reply src/registrar/views/domain.py:139]: You're right, that pass does not make sense. The exception should not happen so maybe that whole try/except block is unneeded and can be replaced by new_email = form.cleaned_data["security_email"]?

**Comment by kewlguy781** [comment]
Look good to me. I have reviewed all the files and codes (thank you for adding the updated migrations file). I would like to see more information under "Context" explaining that security email will be done via EPP instead of DomainInformation, which is why they are seeing these changes. Please add these information, and once you do that, go ahead and merge it.

**Review by kewlguy781 — APPROVED** [review]
Approved with the comment entered previously.

File: src/registrar/views/domain.py (hunk 1 above)
Comment (abroddrick) [reply src/registrar/views/domain.py:135]: I would say, if there is a chance of a crashing exception on the server side, best practice is to catch it. So I wouldn't remove the try/except; for consistency you could follow this pattern where you set new_email to have a default value on error. 
image
or 
just move the domain.set_security_email portion into an else. The else means "If the exception is not hit", so no email gets added if there is no email (which sounds like that shouldn't happen really anyways).
```
       else:
              domain = self.get_object()
              domain.set_security_email(new_email)
```

That being said I saw another part of the code not doing the try/except block too, so I am fine with it being dealers choice and we can circle back to it if we ever need to.

Comment (rachidatecs) [reply src/registrar/views/domain.py:139]: Let's see if 3rd time's the charm. I removed the 'KeyError' on the except as it seemed like it's specific to the Neil's use case and initialized new_email to an empty string.

**Review by neilmb — APPROVED** [review]
This is great work @rachidatecs, very nice to see that these pages are repeatable. I think the only comment that needs following up on is adding the page to the `test_no_domain_role` unit test. 

This is our "saved reply" review checklist. A check means that I checked that the criteria is met or that I checked that the item does not apply.

## Accessibility
- [X] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
   
  We don't have any of these domain detail pages in that file right now, so let's not figure it out on this PR.

- [X] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [X] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [X] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [X] All form inputs have explicit labels
    - [X] All relevant images use an img tag
    - [X] All images have alt attributes
    - [X] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [X] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [X] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [X] Tab order is logical
    - [X] Form instructions are associated with inputs
    - [X] Site doesn’t timeout unexpectedly. Identify elements that may “timeout” and verify that the user can request more time
    - [X] Tables are coded properly. Tables have proper headers and column attributes
    - [X] Headings are nested properly. Heading elements are nested in a logical way
    - [X] Frames are named. All frames have a name element
    - [X] Flashing elements are compliant. Elements that flash on screen do so at a rate of less than 3 Hz
    - [X] Language is set. The language for the page is set
    - [X] The language for sections on the page that differ from the site language are set
    - [X] CSS is not required to use the page. The page makes sense with or without CSS
    - [X] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [X] Page titles are descriptive
    - [X] Required plugins are linked on the page
 
## Device Matrix
- [X] firefox
    - [X] web pages render correctly
    - [X] interactive functionality is operative
- [X] chrome/chromium/edge
    - [X] web pages render correctly
    - [X] interactive functionality is operative
- [X] safari
   
  We have a long-standing bug that Safari doesn't work with Login on localhost, but it appears to work on the sandbox.

- [X] web page is readable and usable 
    - [X] at 480px
    - [X] at 1024px

File: src/registrar/tests/test_views.py [hunk 3]
```diff
@@ -1063,6 +1063,11 @@ def test_not_logged_in(self):
         )
         self.assertEqual(response.status_code, 302)
 
+        response = self.client.get(
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/tests/test_views.py:1066]: I merged too soon. I'll make a follow-up PR

**Pull request merged**